Framework/DataHandling/src/LoadSwans.cpp

```diff
@@ -0,0 +1,230 @@
+#include "MantidDataHandling/LoadSwans.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidDataHandling/LoadHelper.h"
+#include "MantidGeometry/Instrument/ComponentHelper.h"
+
+#include <map>
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+
+namespace Mantid {
+namespace DataHandling {
+
+using namespace Mantid::Kernel;
+using namespace Mantid::Geometry;
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(LoadSwans)
+
+//----------------------------------------------------------------------------------------------
+/** Constructor
+ */
+LoadSwans::LoadSwans() : m_detector_size(0) {
+  m_ws = EventWorkspace_sptr(new EventWorkspace());
```

Comment:
Move this into the initializer list and use `boost::make_shared`. 
It might be cleaner to use C++11 member initialization in the header, then use the implicitly defined default constructor.